CMakeLists.txt

```diff
@@ -109,6 +109,10 @@ add_library(zephyr_interface INTERFACE)
 # flags that come with zephyr_interface.
 zephyr_library_named(zephyr)
 
+if(CONFIG_LEGACY_GENERATED_INCLUDE_PATH)
+  zephyr_include_directories(${PROJECT_BINARY_DIR}/include/generated/zephyr)
```

Comment:
I'm fine with the PR, just one note for discussion (Arch WG): should generated headers have their own subfolder, ie, `<zephyr/generated/xxx.h>`? (maybe with some exceptions like `version.h`).

Reply:
+1, had the same thing in my mind

Reply:
Some will always be generated (e.g. syscall stuff, devicetree, etc.). I think it makes sense to put them into `generated/`, others like `version.h`, probably not.

Reply:
I guess the main concern is that a generated header could conflict with a static header, and that a generated header might incorrectly get included first.

Technically, that's still a risk today, so could be solved separately.